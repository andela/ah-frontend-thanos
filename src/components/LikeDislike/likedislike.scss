$primaryDark: #20c0b0;

.article-like-dislike {
    font-size: 17px;
    margin: 15px 10px;
}
.article-like-dislike {
    .thumbs {
        background: #eaeaea;
        padding: 7px 10px;
    }
    .thumbs:hover {
        cursor: pointer;

        svg path {
            color: $primaryDark;
        }
    }
}