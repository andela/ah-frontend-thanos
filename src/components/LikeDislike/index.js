import React from 'react';
import PropTypes from 'prop-types';
import { library } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faThumbsUp, faThumbsDown,
} from '@fortawesome/free-solid-svg-icons';
import './likedislike.scss';

library.add(faThumbsUp, faThumbsDown);

const createButton = (onClick, icon, cssClass, likeDislikeCount, id) => (
  <React.Fragment key={id}>
    <button type="button" id={id} onClick={onClick} className="thumbs mr-2"><FontAwesomeIcon icon={icon} /></button>
    <span className={cssClass}>{(likeDislikeCount >= 0) ? likeDislikeCount : 0}</span>
  </React.Fragment>
);

const LikeDislike = ({
  onLikeDislike, likes, dislikes, commentId,
}) => {
  const buttons = [
    {
      onClick: (() => { onLikeDislike('like', commentId); }),
      icon: 'thumbs-up',
      cssClass: 'mr-4',
      likeDislikeCount: likes,
      id: 'bt-like',
    },
    {
      onClick: (() => { onLikeDislike('dislike', commentId); }),
      icon: 'thumbs-down',
      cssClass: '',
      likeDislikeCount: dislikes,
      id: 'bt-dislike',
    },
  ];
  return (
    <div className="article-like-dislike">
      <form>
        {buttons.map(
          bt => createButton(bt.onClick, bt.icon, bt.cssClass, bt.likeDislikeCount, bt.id),
        )}
      </form>
    </div>
  );
};

LikeDislike.propTypes = {
  onLikeDislike: PropTypes.func.isRequired,
  likes: PropTypes.number.isRequired,
  dislikes: PropTypes.number.isRequired,
  commentId: PropTypes.number,
};

LikeDislike.defaultProps = {
  commentId: 0,
};

export default LikeDislike;
