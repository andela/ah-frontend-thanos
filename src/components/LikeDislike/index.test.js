import React from 'react';
import { shallow } from 'enzyme';
import LikeDislike from '.';

describe('<LikeDislike />', () => {
  it('renders component', () => {
    const wrapper = shallow(
      <LikeDislike onLikeDislike={jest.fn()} likes={0} dislikes={0} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should default likes/dislikes to 0 if state makes it less than zero', () => {
    const wrapper = shallow(
      <LikeDislike onLikeDislike={jest.fn()} likes={-1} dislikes={-1} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
