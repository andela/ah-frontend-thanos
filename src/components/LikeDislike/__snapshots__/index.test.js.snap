// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LikeDislike /> renders component 1`] = `
<div
  className="article-like-dislike"
>
  <form>
    <button
      className="thumbs mr-2"
      id="bt-like"
      onClick={[Function]}
      type="button"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="thumbs-up"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </button>
    <span
      className="mr-4"
    >
      0
    </span>
    <button
      className="thumbs mr-2"
      id="bt-dislike"
      onClick={[Function]}
      type="button"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="thumbs-down"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </button>
    <span
      className=""
    >
      0
    </span>
  </form>
</div>
`;

exports[`<LikeDislike /> should default likes/dislikes to 0 if state makes it less than zero 1`] = `
<div
  className="article-like-dislike"
>
  <form>
    <button
      className="thumbs mr-2"
      id="bt-like"
      onClick={[Function]}
      type="button"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="thumbs-up"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </button>
    <span
      className="mr-4"
    >
      0
    </span>
    <button
      className="thumbs mr-2"
      id="bt-dislike"
      onClick={[Function]}
      type="button"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="thumbs-down"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </button>
    <span
      className=""
    >
      0
    </span>
  </form>
</div>
`;
